### PR TITLE
Reduce Drive9 metrics and schema diff overhead

### PR DESCRIPTION
**Pull request by srstack**
## Summary
- keep tenant-scoped service operation counters while omitting tenant_id from high-fanout internal service duration histograms
- trust durable tenant schema_version by default for version-matched cold opens instead of periodic schema validation diffs
- parallelize TiDB schema table diff checks with bounded table-level concurrency

## Tests
- make test TEST_PKGS='./pkg/metrics'
- make test TEST_PKGS='./pkg/tenant/schema ./pkg/tenant'

## Summary by CodeRabbit

- **Performance**
  - Improved TiDB schema diffing by processing independent tables concurrently to reduce overall validation time.
- **Bug Fixes**
  - Refined operation metrics labeling so tenant identifiers are included only in the appropriate duration metrics.
  - Avoided redundant TiDB auto-embedding schema work when the tenant schema version already matches the target.
- **Configuration**
  - Disabled periodic TiDB auto-embedding schema validation in favor of version-driven checks.
  - Updated the documented default for `DRIVE9_USER_DB_MAX_IDLE_CONNS` to `0`, and adjusted the default accordingly.

### CODE REVIEW

**Comment by srstack** [comment]
Reviewed head `a6225e83` (read-only diff review). Three independent changes, all reasonable:

**1. Drop tenant_id from high-fanout duration histograms — good cardinality fix.**
`RecordTenantOperation` keeps tenant_id on the *counter* (`serviceOperationsTotal`) but omits it from the *duration histogram* for `file_gc`/`quota_config_cache`/`server_quota`/`user_db_access`. Histograms are the worst cardinality offender (tenant × bucket cross-product), so scoping those four hot components to role-level duration while preserving per-tenant counts is the right trade. The allow-by-default / deny-listed-components shape is clear, and both the omit and keep(`backend`) cases are covered by tests. One consideration: the deny-list is a hardcoded `switch` — if a future high-fanout component is added, it's easy to forget to add it here and silently reintroduce a cardinality spike. A short comment on the switch explaining the selection criterion ("per-tenant duration not worth the cardinality for background/hot-path components") would help the next person decide.

**2. Periodic schema validation default 32 → 0 — behavior change worth confirming.**
This flips version-matched cold opens from "validate 1-in-32" to "trust the durable schema_version, never diff." That's a sound default *if* every schema change goes through a schema_version bump — then a version match genuinely implies schema match. The risk it gives up is detecting **out-of-band drift** (a manual ALTER, a partially-applied migration) where the row's schema_version is unchanged but the actual schema isn't; with sampling off, that now goes unnoticed until something fails at query time. The escape hatch (set the var non-zero in tests/emergency builds) is preserved, and the perf win on cold-open latency is real. Please just confirm the team is comfortable trading the periodic drift-detection safety net for the latency win — that's a product/ops call, not a code issue. Worth a one-line note in ops docs that drift is no longer sampled by default.

**3. Parallelize table diff (bounded worker pool ≤4) — concurrency looks correct.**
- `results[tableIndex]` is written by exactly one worker per index, so the index-sharded writes need no lock — correct, no data race by construction.
- First-error-wins via buffered `errCh` (cap 1) + `select/default` + `cancel()`, then a post-`wg.Wait()` drain of `errCh` and a final `ctx.Err()` recheck — that correctly surfaces both a table-diff error and outer-context cancellation.
- The `sendJobs` loop selects on `diffCtx.Done()` so it stops feeding work once a worker fails. Clean shutdown.
- The concurrency test (blocking on `information_schema.columns` until both tables are observed in-flight) genuinely proves parallelism rather than just asserting a count. Nice.

One thing to sanity-check operationally: this now issues up to 4 concurrent `information_schema` queries per cold open against the tenant's user DB pool. With the user pool default recently raised to 6 open conns (#731), 4 concurrent diff queries fit, but if that pool is smaller in some deploy the diff workers will contend with request-path queries for connections during a cold open. Given periodic validation is now off by default (change #2), the diff path runs far less often, so this is mostly a non-issue — just noting the interaction between the two changes.

Note: I assessed this by reading the diff and the PR's own tests only — I did not compile or run the suite (worktree-based test runs are unreliable in this checkout). The logic and the included tests look correct; recommend confirming CI is green. Modulo the confirm-the-default-change point (#2), LGTM.

File: pkg/metrics/operations.go [hunk 1]
```diff
@@ -117,19 +117,36 @@ func RecordTenantOperation(tenantID, component, operation, result string, d time
 	result = cleanMetricValue(result, "unknown")
 	tenantID = cleanMetricValue(tenantID, "unknown")
 	RegisterModule(component)
-	attrs := []Attribute{
+	counterAttrs := []Attribute{
 		Attr("component", component),
 		Attr("operation", operation),
 		Attr("result", result),
 	}
```
[the remark this answers is not in the document]
Comment (srstack) [reply pkg/metrics/operations.go]: Addressed in 6a12a108: RecordTenantOperation now uses a fixed [4]Attribute base array and derives the counter/duration slices from it, avoiding the duplicate 3-attribute slice literals while preserving the tenant-label behavior.

**Review by mornyx — APPROVED** [review]
## Review — PR #733

**Verdict: APPROVED**

### Summary

Three performance optimizations: (1) omit `tenant_id` from high-fanout internal service duration histograms while keeping counters, (2) trust durable tenant `schema_version` by default for version-matched cold opens instead of periodic schema validation diffs, (3) parallelize TiDB schema table diff checks with bounded concurrency. Also reduces user DB idle connection default from 2 to 0.

### Changes reviewed (8 files, +273/-17)

| Area | Files | Summary |
|------|-------|---------|
| Metrics | `pkg/metrics/operations.go` | `RecordTenantOperation` — counter keeps `tenant_id`, duration histogram omits it for `file_gc`, `quota_config_cache`, `server_quota`, `user_db_access` via `serviceOperationDurationIncludesTenant()`. Fixed-size `[4]Attribute` array avoids heap allocation. |
| Schema validation | `pkg/tenant/pool.go` | `periodicTiDBSchemaValidationEvery` default 32 → 0. `shouldPeriodicValidateTiDBSchemaOnOpen` returns false when 0. |
| Schema diff | `pkg/tenant/schema/tidb_auto.go` | `diffTiDBTables` — fan-out worker pool with `maxConcurrentTiDBSchemaDiffTables=4`, cancel-on-error, pre-allocated results slice |
| DB pool | `pkg/mysqlutil/pool.go`, `cmd/drive9-server/main.go` | `defaultUserMaxIdleConns` 2 → 0 |
| Tests | `operations_test.go`, `pool_test.go`, `tidb_auto_test.go` | Tenant-omit duration test, skip-validation-by-default test, concurrent table diff test |

### My static verification

**`diffTiDBTables` concurrency (my review):**
- Pre-allocated `results [][]tidbSchemaDiff` with `len(tables)` — each index written by exactly one worker, no data race ✓
- `errCh` buffered(1) with `select-default` — first error wins, subsequent dropped ✓
- `cancel()` on error propagates via `diffCtx` to other workers — they stop ranging over `jobs` ✓
- `jobs` channel closed after all sent → workers exit range loop → `wg.Wait()` returns ✓
- Post-wait: checks `errCh` then `ctx.Err()` — correct error propagation ✓
- Concurrency test verifies actual parallelism via blocking mock that requires 2 concurrent queries to proceed ✓

**Schema validation default change (my review):**
- `shouldPeriodicValidateTiDBSchemaOnOpen` returns `false` when `periodicTiDBSchemaValidationEvery == 0` — short-circuits before atomic counter ✓
- Existing test `TestPoolCreateBackendPeriodicallyValidatesSchemaWhenVersionMatches` overrides to 4, still tests the feature ✓
- New test `TestPoolCreateBackendSkipsSchemaValidationWhenVersionMatchesByDefault` verifies default behavior ✓
- Safety: `schema_version` is durable and checked on every cold open; periodic validation was defense-in-depth against out-of-band drift. Setting to 0 is a reasonable latency trade-off. ✓

**Metrics tenant_id omission (my review):**
- Counter always includes `tenant_id` — event counting preserved ✓
- Duration histogram omits `tenant_id` only for 4 high-fanout internal components — reduces cardinality ✓
- `baseAttrs[:3]` / `baseAttrs[:4]` slicing on fixed array — no heap allocation, correct attribute sets ✓
- Test verifies counter has tenant_id, histogram doesn't, for each excluded component ✓
- Test verifies `backend` component still includes tenant_id in histogram ✓

**User DB idle conns 2 → 0 (my review):**
- `SetMaxOpenConns(6)` still allows 6 concurrent connections, all opened on demand ✓
- Eliminates idle connection memory overhead for cached tenant pools ✓
- Trade-off: every operation needs a fresh connection — acceptable for serverless TiDB where idle connections may incur costs ✓

- `go build ./pkg/metrics/... ./pkg/tenant/... ./pkg/mysqlutil/... ./cmd/...` — passes
- CI: CodeRabbit pass, ci pass (6m24s), e2e pass (7m28s)

### Codex review

Codex (glm-5.2, xhigh) ran for the full 500s timeout but did not produce a final conclusion. It was reviewing the entire working tree (42+ files from accumulated PR checkouts) rather than PR #733's 8 files. My static verification covers all key changes in this PR — the concurrency pattern, schema validation default, metrics attribute handling, and DB pool tuning are all correct.

**Review by qiffang — APPROVED** [review]
## Review: GREEN

Three orthogonal optimizations, each correct and well-tested.

### 1. Metrics cardinality reduction (`operations.go`)

`RecordTenantOperation` now omits `tenant_id` from the duration histogram for `file_gc`, `quota_config_cache`, `server_quota`, `user_db_access` — these are high-frequency internal components where per-tenant duration breakdown creates histogram series explosion without actionable insight. The counter still carries `tenant_id` for per-tenant request accounting.

- Fixed-size `[4]Attribute` array avoids heap allocation — clean.
- `serviceOperationDurationIncludesTenant` allowlist is conservative (default `true`). Safe.
- Tests verify both omission (`TestRecordTenantOperationOmitsTenantFromSelectedDurationHistograms`) and retention (`TestRecordTenantOperationKeepsTenantOnBackendDurationHistogram`).

### 2. Parallel schema diff (`tidb_auto.go`)

`diffTiDBTables` runs up to `maxConcurrentTiDBSchemaDiffTables=4` concurrent `diffTiDBTable` calls. Worker pattern is correct:

- `sync.WaitGroup` + channel jobs + results slice indexed by table index (no contention).
- Error cancellation via `context.WithCancel` + first-error-wins `errCh` with capacity 1.
- `sendJobs` loop checks `diffCtx.Done()` to stop sending on error — correct early exit.
- `close(jobs)` after send loop ensures workers drain — no goroutine leak.
- `workerCount` is returned for observability logging.
- Test (`TestDiffTiDBTablesRunsIndependentTablesConcurrently`) verifies actual concurrency by blocking on `information_schema.columns` and requiring both tables to start before releasing.

### 3. User DB idle conns and periodic validation

- `defaultUserMaxIdleConns` changed from 2 → 0: tenant user DB connections are not kept idle in the pool. This reduces persistent connection count across many cached tenants. `sql.DB` will still open connections on demand.
- `periodicTiDBSchemaValidationEvery` changed from 32 → 0: version-matched tenant opens skip schema diff entirely. Schema version is the durable source of truth; periodic sampling added unnecessary cold-open overhead. Test `TestPoolCreateBackendSkipsSchemaValidationWhenVersionMatchesByDefault` verifies no ensure/validate calls when version matches.

**Edge cases checked:**
- `diffTiDBTables` with empty tables → returns `(nil, 0, nil)` ✓
- `diffTiDBTables` with fewer tables than workers → `workerCount = len(tables)` ✓
- Context cancellation mid-diff → workers exit via `diffCtx.Done()` ✓
- CI: `ci` ✅, `drive9-server-local e2e smoke` ✅

**Review by qiffang — APPROVED** [review]
**GREEN** — three orthogonal optimizations, all correct.

## 1. Metrics: omit tenant_id from high-fanout duration histograms

- `RecordTenantOperation` uses fixed `[4]Attribute` array (no allocation). Counter keeps `tenant_id`; duration histogram drops it for `file_gc`, `quota_config_cache`, `server_quota`, `user_db_access`.
- No zero-attribute leak: `baseAttrs[3]` only set when `tenantID != "unknown"`, and `durationAttrs` only extends to `[:4]` when both conditions met.
- Tests verify both omit and keep paths.

## 2. periodicTiDBSchemaValidationEvery = 0

- Disables periodic schema diff on version-matched cold opens. `shouldPeriodicValidateTiDBSchemaOnOpen()` returns false when `every == 0`.
- Design trade-off: no safety net for out-of-band schema drift. Comment says to set non-zero for emergency builds.

## 3. defaultUserMaxIdleConns = 0

- Eliminates idle connection holding for cached tenant user DB pools. Correct trade-off for serverless TiDB billing model.

## 4. diffTiDBTables parallelism

- Worker pool with `maxConcurrentTiDBSchemaDiffTables = 4`. First error cancels remaining via `context.WithCancel`. Results by index (order-preserving).
- Pre-cancelled context safe: `sendJobs` breaks immediately, `ctx.Err()` caught.
- Test deterministically verifies concurrent execution.

**Review by qiffang — APPROVED** [review]
GREEN from my pass on head `d57b57693033368caea523937d761b4e588d6341`.

No blocking findings.

What I checked:
- Metrics cardinality path: `RecordTenantOperation` keeps `tenant_id` on `drive9_service_operations_total` while dropping it from `drive9_service_operation_duration_seconds` only for the four selected high-fanout components. The local registry consumes attrs synchronously via `labelsKey`, so the fixed-array slice reuse is safe. Tests cover both omitted components and the retained `backend` case.
- Schema-version trust path: version-matched tenant opens now skip periodic validation by default, but mismatched versions still run `ensureTiDBSchemaForEmbeddingMode` before recording the target version. This is a deliberate drift-detection tradeoff, not a correctness hole in normal managed schema rollout.
- Parallel schema diff: `diffTiDBTables` bounds parallelism at 4, writes each `results[tableIndex]` from exactly one worker, cancels on first error, stops sending jobs on `diffCtx.Done()`, and returns diffs in stable table order. I do not see a goroutine leak or result race in this pattern.
- User DB idle conns: default idle `0` is reflected in the server help text and leaves `MaxOpenConns` unchanged, so it reduces retained tenant connections without capping request concurrency.

Validation run locally:
- `git diff --check origin/main...HEAD` passed.
- `go test ./pkg/metrics ./pkg/mysqlutil -count=1` passed.
- `go vet ./pkg/metrics ./pkg/mysqlutil ./pkg/tenant ./pkg/tenant/schema` passed.
- `go test -c ./pkg/tenant -o /tmp/drive9-pr733-tenant.test` passed.
- `go test -c ./pkg/tenant/schema -o /tmp/drive9-pr733-tenant-schema.test` passed.
- `go build ./cmd/drive9-server` passed.

I could not execute `go test ./pkg/tenant ./pkg/tenant/schema` on this machine because their `TestMain` starts testcontainers and Docker/rootless Docker is unavailable (`panic: rootless Docker not found`). GitHub `ci` and `drive9-server-local e2e smoke` are green on the PR.

**Review by mornyx — APPROVED** [review]
## Review — PR #733 (round 2)

**Verdict: APPROVED**

Head updated from `d57b5769` to `5a42c956` (+4 commits, +2009/-310 lines across 22 files). This round reviews the new changes since the first review.

### New changes since round 1

**1. Tenant pool idle reaper (`pkg/tenant/pool.go`)**
- New `IdleTimeout` / `IdleReapInterval` config fields + `Start(ctx)` / `reapLoop` / `reapOnce` goroutine
- `entry.lastUsed` field refreshed by `Get` / `Acquire` / `S3Backend` (foreground + durable work) and on `Acquire` release
- `AcquireCached` (safety-net scan) uses `makeReleaseNoRefresh` — does NOT refresh idle timer, so internal scans can't keep idle tenants warm forever
- `Close()` stops reaper first (via `reapStop` + `reapWG.Wait()`) to prevent race with shutdown eviction
- Periodic schema validation fully removed (`shouldPeriodicValidateTiDBSchemaOnOpen` deleted, `validateTiDBSchemaForEmbeddingMode` deleted)

**My verification:** Reaper correctly skips `retired` and `refs > 0` entries. `lastUsed` is set under `p.mu` lock on all access paths. `makeReleaseNoRefresh` vs `makeRelease` split is correct — AcquireCached doesn't refresh, all other paths do. Close-then-reap ordering prevents eviction races. 10 new tests cover idle eviction, refs, disabled, AcquireCached non-refresh, Start/Close lifecycle. ✓

**2. Write generation guard for shadow/pendingIndex cleanup (`pkg/fuse/shadow.go`, `pending_index.go`, `writeback_uploader.go`, `dat9fs.go`)**
- New `writeGen` map in `ShadowStore` — bumped on every mutating write (`WriteFull`, `WriteAt`, `Ensure`, `WriteStream`, `Truncate`, `WriteExtents`)
- `RemoveIfGeneration(path, expectedGen)` — atomically checks gen under lock before removing; returns false if gen mismatch (newer write happened)
- `PendingIndex.RemoveIfGeneration` — same pattern
- `StagingGens` struct captures writeBack + pendingIndex + shadowStore generations at upload start via `GetMetaAndViewWithCallback` (callback runs under path lock for atomic snapshot)
- `onWriteBackUploadSuccess` uses `RemoveIfGeneration` with captured gens — stale upload can't remove fresher staging entry

**My verification:** This is a real concurrency fix. Previously, a stale upload's cleanup could `Remove` a shadow/pendingIndex entry that was created by a newer same-path write during the upload. The `writeGen` + `RemoveIfGeneration` pattern makes the check-and-remove atomic under `s.mu`. `snapshotStagingGens` deliberately avoids calling `writeBack.GetMeta` (would self-deadlock on path lock) — WriteBackGen is set by caller from `meta.Generation`. 7 new tests including race tests (`TestShadowStoreRemoveIfGenerationRaceStaleVsNewerWrite`, `TestOnWriteBackUploadSuccessStaleUploadPreservesNewerPending`). ✓

**3. Flush debouncer concurrency hardening (`pkg/fuse/debounce.go`)**
- `inflight` map tracks executing callbacks per path
- `pendingFlush.done` channel + `once` for safe completion signaling
- `Cancel` now blocks until in-flight callback finishes (via `<-pf.done`)
- `runCallback` verifies pointer identity of `pendingFlush` before executing
- No double-inflight for same path (timer waits for existing inflight)

**My verification:** At most one callback per path executing at any time. Stale timer can't clobber newer entry (pointer identity check). Cancel blocks for safety (Unlink can inspect server state after). 8 new tests cover cancel-waits, schedule-replaces-while-running, no-double-inflight, stale-timer, flushall. ✓

**4. SSE establishment duration (`pkg/server/instrumentation.go`)**
- SSE streams now record time-to-first-byte (request → 200 headers) into HTTP duration histograms instead of skipping them
- Full connection lifetime still goes to dedicated `drive9_sse_connection_duration_seconds`
- `sseEstablishmentDuration` returns bounded duration for observe

**My verification:** `markSSEStreamEstablished` records `sseEstablishedAt` under mutex. `observe` uses `sseEstablishmentDuration` (bounded) instead of `time.Since(start)` (unbounded). Test verifies duration is bounded and < connection lifetime. ✓

**5. `WaitPrefixTimeout` (`pkg/fuse/commit_queue.go`)**
- Bounded version of `WaitPrefix` for Rmdir — polls in-flight + queue with 50ms sleep, returns false on timeout

**6. Schema context (`pkg/tenant/schema/context.go`)**
- `WithTenantID` / `tenantIDFromContext` — passes tenant ID through schema-init context for user_schema DB pool metrics

**7. Other**: User DB idle conns 2→0, `cmd/drive9-server/main.go` help text update, `tidb_auto.go` minor fix

### Verification summary

- `go build ./pkg/tenant/... ./pkg/server/... ./pkg/fuse/... ./pkg/metrics/...` — passes
- CI: CodeRabbit pass, ci pass (6m20s), e2e pass (7m28s)
- 29 new tests across pool, fuse, instrumentation, schema
- Codex (glm-5.2, xhigh): ran 500s, did not produce final conclusion (was reviewing full working tree + attempting test runs that need Docker). My static verification covers all key changes.

### Assessment

The writeGen guard and debouncer hardening are real concurrency safety improvements that fix potential data-loss races in the writeback upload path. The idle reaper is well-designed with correct foreground-vs-scan attribution. SSE establishment duration is a clean metrics improvement. No blockers found.

**Comment by qiffang** [comment]
## Post-merge independent review — REVISE / follow-up required

Reviewed the exact final PR range `927f529...5a42c956` (10 files), not the accumulated branch history. Metrics cardinality handling and bounded schema-diff concurrency look correct. I found one production-risk default change and two correctness/coverage gaps that should be followed up after the already-completed merge.

### B1: `defaultUserMaxIdleConns = 0` changes every independent DB call into a new physical connection

`pkg/mysqlutil/pool.go:17-19` sets the user pool to `MaxIdleConns=0`. In Go `database/sql`, this means **no returned connection is retained**; `ConnMaxIdleTime=15s` is therefore ineffective. `MaxOpenConns=6` preserves concurrency, but it does not preserve connection reuse.

This is more than “drop persistent idle connections.” Sequential non-transactional queries now each pay a fresh TCP/TLS/MySQL connection setup. For example, `pkg/datastore/store.go:1675-1689` implements a normal file `Stat` as `GetNode` followed by `GetFile`; with max idle 0 those two sequential queries cannot reuse one connection. FUSE/API paths containing multiple store calls amplify this further.

I verified the `database/sql` behavior with a counting driver: two sequential `PingContext` calls opened 2 physical connections at `MaxIdleConns=0`, versus 1 at `MaxIdleConns=2`.

The added tests only assert constants. `pkg/server/testmain_test.go:14-20` already forces `DRIVE9_USER_DB_MAX_IDLE_CONNS=0`, so CI proves correctness under churn but does not measure the production latency/connection-open-rate regression introduced by changing the default from 2 to 0. The PR body also does not call out or provide rollout evidence for this production default change.

Before broad rollout, either restore a small idle pool with the existing short idle TTL, or provide production/representative evidence for connection-open rate, P50/P99 latency, and TiDB connection cost that explicitly accepts the per-query reconnect trade-off.

### B2: the schema-version comment now states a false invariant

`pkg/tenant/pool.go:868-870` says any schema divergence between consecutive opens will be caught on the next open because the stored version will differ. After this PR, version-matched opens never inspect the physical schema, so an out-of-band/manual ALTER with unchanged `schema_version` is **not** caught on the next open (or any later open).

Trusting the durable version may be an intentional latency policy, but code/docs must state the real boundary: only a target-version change triggers ensure; version-preserving physical drift is undetected until a query fails or an explicit validation/repair path is run. Please correct the comment and add the operational detection/recovery contract.

### B3: tenant attribution is missing on synchronous fork schema initialization

`pkg/server/server.go:5347-5350` adds `tenantschema.WithTenantID` only inside `initTenantSchemaAsync`. The synchronous fork path calls `schemaInitForTenant(...)(ctx, dsn)` directly at `pkg/server/fork.go:635`, so `OpenTiDBSchemaDB` at `pkg/tenant/schema/tidb_auto.go:1376-1380` registers `user_schema` metrics with an empty/unknown tenant for fork initialization.

This is not a regression from pre-PR behavior, but it makes the new “tenant label on user schema pool metrics” behavior incomplete. Add tenant context at the common schema-init boundary (or the fork call site) and a regression covering both async provision and fork paths.

### Validation

- `git diff --check 927f529...5a42c956` — pass
- `go test ./pkg/metrics ./pkg/mysqlutil -count=1` — pass
- `go test -race ./pkg/metrics ./pkg/mysqlutil -count=1` — pass
- `go vet ./pkg/metrics ./pkg/mysqlutil ./pkg/tenant ./pkg/tenant/schema ./pkg/server ./cmd/drive9-server` — pass
- compile-only tests for `pkg/tenant`, `pkg/tenant/schema`, `pkg/server`, and `cmd/drive9-server` — pass
- GitHub CI/e2e/CodeRabbit — green on final head

I would not revert the metrics or parallel-diff changes. B1 needs an explicit performance/cost decision with evidence; B2/B3 need focused follow-ups.

**Comment by srstack** [comment]
Follow-up created: #736.

- B1 (`defaultUserMaxIdleConns=0`): keeping this unchanged. The no-idle user DB pool default is intentional for connection-retention control, and deployments can still override it with `DRIVE9_USER_DB_MAX_IDLE_CONNS` if they want connection reuse.
- B2: #736 clarifies the schema-version trust boundary: version-matched opens trust the durable schema version and do not detect out-of-band physical drift unless the target version changes or an explicit validation/repair path is run.
- B3: #736 moves tenant context attribution into `schemaInitForTenant`, so synchronous fork schema initialization also carries tenant attribution for `user_schema` pool metrics, with regression coverage.
